cmd/scollector/collectors/rabbitmq.go

```diff
@@ -0,0 +1,691 @@
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+	res, err := client.Get(url + "/api/overview")
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+	content, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+	var overview map[string]interface{}
+	if err := json.NewDecoder(bytes.NewReader(content[:])).Decode(&overview); err != nil {
+		return nil, err
+	}
+	flatOverview, err := explodeMap(overview, "", ".")
+	if err != nil {
+		return nil, err
+	}
+	var md opentsdb.MultiDataPoint
+	for k, v := range flatOverview {
+		if m, ok := rabbitmqOverviewMeta[k]; ok {
+			val, ok := v.(float64)
+			if !ok {
+				err := fmt.Errorf("rabbitmq: unexpected content or type for '%s' key", k)
+				return nil, err
+
+			}
+			tags := opentsdb.TagSet{}
+			Add(&md, "rabbitmq."+m.Metric, val, tags, m.RateType, m.Unit, m.Desc)
+		}
+	}
+	return md, nil
+}
+
+func rabbitmqQueues(url string) (opentsdb.MultiDataPoint, error) {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+	queueRes, err := client.Get(url + "/api/queues")
+	if err != nil {
+		return nil, err
+	}
+	defer queueRes.Body.Close()
+	queueContent, err := ioutil.ReadAll(queueRes.Body)
+	if err != nil {
+		return nil, err
+	}
+	var queues []map[string]interface{}
+	if err := json.NewDecoder(bytes.NewReader(queueContent[:])).Decode(&queues); err != nil {
+		return nil, err
+	}
+	var md opentsdb.MultiDataPoint
+	for _, qv := range queues {
+		queueName, ok := qv["name"].(string)
+		if !ok {
+			err := fmt.Errorf("rabbitmq: invalid queue name '%s'", queueName)
+			return nil, err
+		}
+		if strings.HasPrefix(queueName, "amq.gen") {
+			// skip generated queues
+			continue
+		}
+		tags := opentsdb.TagSet{"queue": queueName}
+		for k, v := range qv {
+			if m, ok := rabbitmqQeuesMeta[k]; ok {
+				val, ok := v.(float64)
+				if !ok {
+					err := fmt.Errorf("rabbitmq: unexpected content or type for '%s' key", k)
+					return nil, err
+
+				}
+				Add(&md, "rabbitmq.queue."+m.Metric, val, tags, m.RateType, m.Unit, m.Desc)
+			}
+		}
+	}
+	return md, nil
+}
+
+func rabbitmqNodes(url string) (opentsdb.MultiDataPoint, error) {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+	res, err := client.Get(url + "/api/nodes")
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+	content, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+	var nodes []map[string]interface{}
+	if err := json.NewDecoder(bytes.NewReader(content[:])).Decode(&nodes); err != nil {
+		return nil, err
+	}
+	var md opentsdb.MultiDataPoint
+	for _, nv := range nodes {
+		name, ok := nv["name"].(string)
+		if !ok {
+			err := fmt.Errorf("rabbitmq: invalid node name '%s'", name)
+			return nil, err
+		}
+		if !strings.HasSuffix(name, util.Hostname) {
```

Comment:
`/api/nodes` exposes metrics for all nodes in cluster, given that scollector runs on each I just skip everything that's not for current node.